Mage/src/main/java/mage/filter/common/FilterCompareManaValue.java

```diff
@@ -0,0 +1,34 @@
+package mage.filter.common;
+
+import mage.abilities.Ability;
+import mage.abilities.dynamicvalue.DynamicValue;
+import mage.abilities.dynamicvalue.common.StaticValue;
+import mage.cards.Card;
+import mage.constants.ComparisonType;
+import mage.filter.FilterCard;
+import mage.game.Game;
+
+import java.util.UUID;
+
+public class FilterCompareManaValue extends FilterCard {
```

Comment:
Is there a reason you can't just use `ManaValuePredicate` rather than creating a new filter class?

Reply:
I looked at `ManaValuePredicate` but the problem with predicates, as far as I can tell, is that they can't access the source `Ability`, so you can't evaluate a DynamicValue in them. If there is a way to get it, I'd love to know how/an example and implement it that way.

Reply:
Basically, the DynamicValue needs to be evaluated within an `apply()` method of a custom effect class like `int xValue = SavedDamageValue.MUCH.calculate(game, source, this);` Then the filter can also be created within that custom effect using the calculated value.

I don't have a perfect reference but Blast Zone shows some of it.